Mathlib/Algebra/Module/Zlattice.lean

```diff
@@ -261,3 +300,140 @@ protected theorem isAddFundamentalDomain [Finite ι] [MeasurableSpace E] [OpensM
 end Real
 
 end Zspan
+
+section Zlattice
+
+open Submodule
+
+variable (K : Type _) [NormedLinearOrderedField K] [HasSolidNorm K] [FloorRing K]
+variable {E : Type _} [NormedAddCommGroup E] [NormedSpace K E] [FiniteDimensional K E]
+variable {L : AddSubgroup E}
+variable (hd : ∀ r : ℝ, ((L : Set E) ∩ (Metric.closedBall 0 r)).Finite)
+variable (hs : span K (L : Set E) = ⊤)
```

Comment:
How about using:
```suggestion
variable [DiscreteTopology L] (hs : span K (L : Set E) = ⊤)
```
You might have to wait for #5969 to land but I think it might make for a more useful theory.

Reply:
Yes, that's a good idea. It was something I was planning to do and then forgot. I'll have a look soon. Thanks

Reply:
@ocfnash I have committed a new version that uses `DiscreteTopology`. In the end, I did not need the results of #5969 but the much more basic result that the intersection of a bounded set and a closed discrete set is finite that I added to `Topology.MetricSpace.Basic`